File: Mario/mario.c
#include <cs50.h>
#include <stdio.h>

void build(int x);

int main(void)
{
    // Prompting user for integer
    int Height = get_int("Height:\n");

    // Checking if input between 1 - 8
    if (Height >= 1 && Height <= 8)
    {
        build(Height);
    }
    else
    {
        main();
    }
}
// Building function
void build(int x)
{
    for (int i = 1; i <= x; i++)
    {
        // adding space every new line corresponds to wide
        for (int n = x - i; n > 0; n--)
        {
            printf(" ");
        }

        // building left blocks corresponding to wide
        for (int j = 1; j <= i; j++)
        {
            printf("#");
        }

        // printing in between space
        printf("  ");

        // building left blocks corresponding to wide
        for (int k = 1; k <= i; k++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: Mario Python/mario.py
from cs50 import get_int

# getting height from user
h = get_int("Hieght: ")

# prompting user for height if entered invalid value
while h < 1 or h > 8:
    h = get_int("Hieght: ")

# bulding the pyramids
for i in range(1, h+1):
    
    # printing pre-spaces for left side pyramid
    for r in range(h - i):
        print(" ", end='')
    
    # building left-side pyramid
    for j in range(1, i+1):
        print("#", end='')
    
    # printing in-between spaces
    print("  ", end='')

    # printing right-side pyramid
    for k in range(1, i+1):
        print("#", end='')
    
    # ending each line with line break
    print("")